app/domain/authentication/authn_oidc/authenticate_id_token/fetch_provider_certificate.rb

```diff
@@ -0,0 +1,44 @@
+require 'uri'
+require 'json'
+
+module Authentication
+  module AuthnOidc
+    module AuthenticateIdToken
+      FetchProviderCertificate = CommandClass.new(
+        dependencies: {
+          logger: Rails.logger
+        },
+        inputs: %i(provider_uri)
+      ) do
+
+        def call
+          # provider discovery might throw exception. Let it propagate upward
+          discover_provider
+          fetch_certs
+        end
+
+        private
+
+        def discover_provider
+          @logger.debug("[OIDC] Discovering provider '#{@provider_uri}'")
+
+          @discovered_provider = OpenIDConnect::Discovery::Provider::Config.discover!(@provider_uri)
+        rescue HTTPClient::ConnectTimeoutError => e
+          raise ProviderDiscoveryTimeout.new(@provider_uri, e.inspect)
```

Comment:
You could silence this `reek` error by creating a `raise_error` method:

```
def raise_error(error_class, original_error)
  raiser error_class.new(@provider_uri, original_error.inspect)
end
```

Reply:
done